### PR TITLE
Always download coredistools package

### PR DESCRIPTION
**Pull request by sejongoh**
runtest.sh/cmd currently downloads the coredistools package if COMPlus_GCStress variable is set by TestEnv or test-env options. This is a bit complicated and this doesn't work with a new option for GC stress testing in build/buildtest.cmd. This changeset moves downloading the coredistools package to buildtest.cmd where it takes gcstress option and downloads other packages.

Linux build doesn't have buildtest.sh, so package donwloading is added to build.sh.

### CODE REVIEW

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT Checked gcstress15_pri1r2r

**Comment by adityamandaleeka** [comment]
@sejongoh Can you provide a quick explanation of why this is being done? It looks like we're adding yet another package restore step to build.sh, which we should avoid unless it's really necessary.

**Comment by sejongoh** [comment]
@adityamandaleeka On Windows, downloading coredistools is being moved to buildtest.cmd. Linux build has only build.sh and runtests.sh. build.sh already has network dependence but runtest.sh doesn't seem to have any except this one. Is there a better place to put this?

**Comment by sejongoh** [comment]
@adityamandaleeka FYI, this is moved from runtest.sh. I missed adding runtest.sh into this PR. We can leave it in runtest.sh if it is preferred.

**Comment by sejongoh** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test FreeBSD x64 Checked Build

**Comment by sejongoh** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT Checked gcstress15_pri1r2r

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT gcstress0xc

**Comment by sejongoh** [comment]
@dotnet-bot test this please

**Comment by adityamandaleeka** [comment]
@sejongoh Would it be possible to put it under an argument in build.sh so it doesn't happen by default?

File: build.sh [hunk 1]
```diff
@@ -293,7 +293,15 @@ build_mscorlib()
     fi 
 }
 
-
+resolve_runtime_depedendence()
```
Comment (adityamandaleeka) [inline build.sh]: dependencies

File: build.sh [hunk 2]
```diff
@@ -663,6 +671,10 @@ build_coreclr
 
 build_mscorlib
 
+# Resolve runtime dependences
```
Comment (adityamandaleeka) [inline build.sh]: 'dependencies' here too (and everywhere else)

**Comment by sejongoh** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test Windows_NT x64 Debug Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test Ubuntu x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test  Windows_NT x64 Release Priority 1 Build and Test

**Comment by sejongoh** [comment]
@dotnet-bot test OSX x64 Checked Build and Test

File: tests/runtest.sh [hunk 3]
```diff
@@ -866,35 +866,11 @@ else
...
-# Download runtime dependent libraries
-if [ ! -z "$GCStressLevel" ]; then
-    ${scriptPath}/setup-runtime-dependencies.sh --outputDir=$coreOverlayDir
-    if [ $? -ne 0 ] 
-    then
-        echo 'Failed to download coredistools library'
-        exit $EXIT_CODE_EXCEPTION
-    fi
+${scriptPath}/setup-runtime-dependencies.sh --outputDir=$coreOverlayDir
+if [ $? -ne 0 ]
```
Comment (RussKeldorph) [inline tests/runtest.sh]: I realize one is inside runtest and the other is inside buildtest, but I'm not a fan of the additional inconsistency that for bash we check the return code and cmd we don't.  I would probably just make sure both setup-runtime-dependencies print meaningful error messages on all the failure paths and remove this part here.

**Comment by RussKeldorph** [comment]
@sejongoh What's holding this up?

**Comment by sejongoh** [comment]
I will merge this soon. Please let me know if you have any comment.

**Pull request merged**